### PR TITLE
koord-manager: add unallocated resource into mid resource.

### PR DESCRIPTION
**Pull request by tan90github**
### Ⅱ. Does this pull request fix one issue?
add unallocated resource into mid resource. [refer link](https://koordinator.sh/docs/designs/node-prediction#mid-tier-overcommitment)

### V. Checklist

- [ ] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

**Comment by tan90github** [comment]
@j4ckstraw @yangfeiyu20102011

File: pkg/slo-controller/noderesource/plugins/midresource/plugin.go
Comment (saintube) [inline pkg/slo-controller/noderesource/plugins/midresource/plugin.go:192]: Please consider individual parameters in the ColocationConfig to over-commit the Prod unallocated resources. Since the Prod pods may schedule onto the node anytime, the Prod unallocated resources are often rather unstable for Mid pods than the reclaimable resources.

**Comment by saintube** [comment]
@tan90github BTW, please rebase to the main branch to resolve the E2E issue.

File: pkg/slo-controller/noderesource/plugins/midresource/plugin.go
Comment (tan90github) [reply pkg/slo-controller/noderesource/plugins/midresource/plugin.go:206]: done.

File: pkg/util/sloconfig/colocation_config.go
Comment (saintube) [inline pkg/util/sloconfig/colocation_config.go:73]: Currently the unallocated for Mid tier is still unstable. So I suggest setting the ratio to zero by default and enabling it when we know how to handle the risk.

Comment (tan90github) [reply pkg/util/sloconfig/colocation_config.go]: fixed.

**Comment by tan90github** [comment]
All conversations are resolved. PTAL @saintube , thanks.

File: pkg/slo-controller/noderesource/plugins/midresource/plugin.go
Comment (saintube) [inline pkg/slo-controller/noderesource/plugins/midresource/plugin.go:206]: how about `unallocatedMilliCPU` or `unallocatedMilliCores`, let's make the naming concise

File: pkg/slo-controller/noderesource/plugins/midresource/plugin.go
Comment (saintube) [inline pkg/slo-controller/noderesource/plugins/midresource/plugin.go:269]: for the current usage, it better return zero

File: pkg/slo-controller/noderesource/plugins/midresource/plugin.go
Comment (tan90github) [reply pkg/slo-controller/noderesource/plugins/midresource/plugin.go]: pretty good, fixed.

File: pkg/slo-controller/noderesource/plugins/midresource/plugin.go
Comment (tan90github) [reply pkg/slo-controller/noderesource/plugins/midresource/plugin.go]: fixed.

**Comment by saintube** [comment]
/lgtm
PTAL /cc @zwzhang0107 @jasonliu747

**Comment by tan90github** [comment]
Do you have any other questions about this PR? @zwzhang0107 @jasonliu747 PTAL, thanks.

**Review by jasonliu747 — APPROVED** [review]
/lgtm

**Comment by tan90github** [comment]
After rebase, you may need to re-Approve to merge the PR? Thank you very much! @jasonliu747

**Review by jasonliu747 — APPROVED** [review]
/lgtm

**Comment by tan90github** [comment]
@jasonliu747  thanks for your replay, but it seems that the koordinator-bot did not come to mark the approve label. is it a bug?

**Comment by tan90github** [comment]
> @jasonliu747 thanks for your replay, but it seems that the koordinator-bot did not come to mark the approve label. is it a bug?

@saintube @jasonliu747 This PR has been Approved for a long time, but it has not been merged. Please remind me when it can be merge and approve after I rebase the code. Thanks!

**Comment by saintube** [comment]
@tan90github Since there is an API change in this PR, it requires the approval of the api-machinery OWNER.
PTAL /cc @zwzhang0107 @hormes

**Pull request merged**
